adapters/src/main/java/io/realm/RealmBaseAdapter.java

```diff
@@ -86,6 +89,7 @@ private void removeListener(OrderedRealmCollection<T> data) {
      * @return the number of items.
      */
     @Override
+    @CheckResult
```

Comment:
I understand @cmelchior's argument. But not using the return value indicates that the programmer has made a mistake, and `@CheckResult` can help her to realise it quickly.